src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -333,23 +333,33 @@ private void LookupForMatchingGitHubRepository(BuildData mergedBuild)
 
             using (var client = new HttpClient())
             {
-                string[] segments = mergedBuild.AzureDevOpsRepository.Split('/');
-                string repoName = segments[segments.Length - 1];
-                int index = repoName.IndexOf('-');
+                Uri repoAddr = new Uri(mergedBuild.AzureDevOpsRepository);
+                string repoIdentity = string.Empty;
 
-                StringBuilder builder = new StringBuilder(repoName);
-                builder[index] = '/';
+                if (repoAddr.Host.Equals("github.com", StringComparison.OrdinalIgnoreCase))
+                {
+                    repoIdentity = repoAddr.AbsolutePath.TrimStart('/').TrimEnd('/');
+                }
+                else
+                {
+                    string[] segments = mergedBuild.AzureDevOpsRepository.Split('/');
+                    string repoName = segments[segments.Length - 1];
+                    int index = repoName.IndexOf('-');
+
+                    StringBuilder builder = new StringBuilder(repoName);
+                    builder[index] = '/';
 
-                repoName = builder.ToString();
+                    repoIdentity = builder.ToString();
+                }
 
                 client.BaseAddress = new Uri("https://api.github.com");
                 client.DefaultRequestHeaders.Add("User-Agent", "PushToBarTask");
 
-                HttpResponseMessage response = client.GetAsync($"/repos/{repoName}/commits/{mergedBuild.Commit}").Result;
+                HttpResponseMessage response = client.GetAsync($"/repos/{repoIdentity}/commits/{mergedBuild.Commit}").Result;
 
                 if (response.IsSuccessStatusCode)
```

Comment:
>IsSuccessStatusCode [](start = 29, length = 19)

If we get to set the GitHubRepository, should we set AzureDevOpsRepository (and maybe other AzDO properties) to null?

Reply:
As long as `publishUsingPipelines` is false, pipeline publishing won't use any of those properties as it won't be executed.